src/bios_handler.cpp

```diff
@@ -362,8 +365,91 @@ void IbmBiosHandler::processCreateDefaultLpar()
         saveCreateDefaultLparToBios(*l_pVal);
         return;
     }
-
     logging::logMessage(
         "Invalid type recieved for create default Lpar from VPD.");
 }
+
+void IbmBiosHandler::saveClearNvramToVpd(const std::string& i_clearNvramVal)
+{
+    if (i_clearNvramVal.empty())
+    {
+        logging::logMessage(
+            "Empty value received for clear NVRAM from BIOS. Skip updating to VPD.");
+        return;
+    }
+
+    // Read required keyword from DBus as we need to set only a Bit.
+    auto l_kwdValueVariant = dbusUtility::readDbusProperty(
+        constants::pimServiceName, constants::systemVpdInvPath,
+        constants::utilInf, constants::kwdClearNVRAM_CreateLPAR);
+
+    if (auto l_pVal = std::get_if<std::string>(&l_kwdValueVariant))
+    {
+        types::BinaryVector l_valToUpdateInVpd;
+        if (i_clearNvramVal.compare("Enabled") == constants::STR_CMP_SUCCESS)
+        {
+            // 3rd bit is used to store the value.
+            l_valToUpdateInVpd.emplace_back((*l_pVal).at(0) | 0x04);
+        }
+        else
+        {
+            // 3rd bit is used to store the value.
+            l_valToUpdateInVpd.emplace_back((*l_pVal).at(0) & ~(0x04));
+        }
+        // TODO: Write API to be called to update in VPD.
+        return;
+    }
+    logging::logMessage("Invalid type recieved for clear NVRAM from VPD.");
+}
+
+void IbmBiosHandler::saveClearNvramToBios(const std::string& i_clearNvramVal)
+{
+    // checking for exact length as it is a string and can have garbage value.
```

Comment:
nit: grammer 
Check for the exact length as it is a string and it can have a garbage value.